### ISSUE TITLE
Enabling slf based logging breaks remote debugging

### ISSUE THREAD

**Issue by ChrisHuebsch-FLIG** [eclipse/jetty.project#5895]
**Jetty version** 9.4.32.v20200930

**Java version** 
openjdk version "1.8.0_262"
OpenJDK Runtime Environment (Zulu 8.48.0.52-SA-win64) (build 1.8.0_262-b19)
OpenJDK 64-Bit Server VM (Zulu 8.48.0.52-SA-win64) (build 25.262-b19, mixed mode)

**Question**
I develop a soap webservice using jetty.
I use eclipse and remote debugger to debugg my code.
Recently I enabled the module "logging-jul" which relies on slf-logging.

With this module enabled, breakpoints in my code no longer trigger.

I also noticed, that the list of thrads somehow seems to differ.

(For simplicity I use the standard jdb-console debugger for thread list)

Without logging-jul I have this list of threads shortly after start:
```> threads
Group system:
  (java.lang.ref.Reference$ReferenceHandler)0x14f Reference Handler     cond. waiting
  (java.lang.ref.Finalizer$FinalizerThread)0x14e  Finalizer             cond. waiting
  (java.lang.Thread)0x14d                         Signal Dispatcher     running
  (java.lang.Thread)0x14c                         Attach Listener       running
Group main:
  (java.lang.Thread)0x1                           main                  running
  (java.lang.Thread)0x705                         ShutdownMonitor       running
  (java.lang.Thread)0x721                         qtp26117480-16        cond. waiting
  (java.lang.Thread)0x722                         qtp26117480-17        cond. waiting
  (java.lang.Thread)0x723                         qtp26117480-18        cond. waiting
  (java.lang.Thread)0x724                         qtp26117480-19        cond. waiting
  (java.lang.Thread)0x726                         qtp26117480-20        cond. waiting
  (java.lang.Thread)0x727                         qtp26117480-21        cond. waiting
  (java.lang.Thread)0x729                         qtp26117480-22        cond. waiting
  (java.lang.Thread)0x728                         qtp26117480-23        cond. waiting
  (java.lang.Thread)0x72a                         qtp26117480-24        cond. waiting
  (java.lang.Thread)0x72b                         qtp26117480-25        cond. waiting
  (java.lang.Thread)0x799                         Thread-11             running
  (java.lang.Thread)0x7aa                         PathWatcher@50d0686   cond. waiting
  (java.lang.Thread)0x892                         Scheduler-757108857-1 cond. waiting
```

Enabling logging-jul it looks like this:

```
Group system:
  (java.lang.ref.Reference$ReferenceHandler)0x14f Reference Handler cond. waiting
  (java.lang.ref.Finalizer$FinalizerThread)0x14e  Finalizer         cond. waiting
  (java.lang.Thread)0x14d                         Signal Dispatcher running
  (java.lang.Thread)0x14c                         Attach Listener   running
Group main:
  (java.lang.Thread)0x1                           main              running
  (java.lang.Thread)0x4ac                         Thread-1          running
  (java.lang.Thread)0x4ad                         Thread-2          running
  (java.lang.Thread)0x4ae                         Thread-3          running
```

Where did all those qtp-threads went? Is this behaviour expected?
Can slf logging and debugging be used together?

Surprisingly, they are still there, when looking with Oracles VisualVM.

**Comment by joakime** [comment]
On Jetty 9.4.x, if you enable any of the logging modules with start.jar, then the JVM forks to satisfy the needs of the logging framework.
To enable remote debugging, you'll have to pass the JVM args to the forked JVM.
See the `jvm` module, and know that you'll have to add those JVM args to the content in your `${jetty.base}/start.d/jvm.ini`.

This is the reason your breakpoints don't trigger, as you have only enabled remote debugging on the same JVM that start.jar is running in, not the forked JVM that has the server + logging enabled.

This forking behavior on logging has been fixed in Jetty 10.0.x

**Comment by ChrisHuebsch-FLIG** [comment]
Awesome! Thank you so much for your anwer.
It works!

In case someone else is reading this ticket... put this in jvm.ini:

```
--exec 
-Xdebug
-agentlib:jdwp=transport=dt_socket,address=8998,server=y,suspend=n
```

Each one on its own line.
